pkg/drawio/squareTreeNode.go

```diff
@@ -0,0 +1,162 @@
+	return abstractSquareTreeNode{abstractTreeNode: newAbstractTreeNode(parent, name)}
+}
+func (tn *abstractSquareTreeNode) addIconTreeNode(icon IconTreeNodeInterface) {
+	tn.elements = append(tn.elements, icon)
+}
+func (tn *abstractSquareTreeNode) addLineTreeNode(line LineTreeNodeInterface) {
+	tn.connections = append(tn.connections, line)
+}
+
+func (tn *abstractSquareTreeNode) IconTreeNodes() []IconTreeNodeInterface {
+	return tn.elements
+}
+func (tn *abstractSquareTreeNode) IsSquare() bool { return true }
+func (tn *abstractSquareTreeNode) TagID() uint    { return tn.id + 1 }
+
+func (tn *abstractSquareTreeNode) setGeometry() {
+	location := tn.Location()
+	parentLocation := location
+	if tn.Parent() != nil {
+		parentLocation = tn.Parent().Location()
+	}
+	tn.width = location.lastCol.width() + location.lastCol.x() - location.firstCol.x()
+	tn.hight = location.lastRow.hight() + location.lastRow.y() - location.firstRow.y()
+	tn.x = location.firstCol.x() - parentLocation.firstCol.x()
+	tn.y = location.firstRow.y() - parentLocation.firstRow.y()
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+type NetworkTreeNode struct {
+	abstractSquareTreeNode
+	vpcs []SquareTreeNodeInterface
+}
+
+func NewNetworkTreeNode() *NetworkTreeNode {
+	return &NetworkTreeNode{}
+}
+
+func (tn *NetworkTreeNode) children() ([]SquareTreeNodeInterface, []IconTreeNodeInterface, []LineTreeNodeInterface) {
+	return tn.vpcs, tn.elements, tn.connections
+}
+func (tn *NetworkTreeNode) IsNetwork() bool { return true }
+
+// ////////////////////////////////////////////////////////////////////////////////////////
+type VpcTreeNode struct {
+	abstractSquareTreeNode
+	zones []SquareTreeNodeInterface
+	sgs   []SquareTreeNodeInterface
+}
+
+func NewVpcTreeNode(parent *NetworkTreeNode, name string) *VpcTreeNode {
+	vpc := VpcTreeNode{abstractSquareTreeNode: newAbstractSquareTreeNode(parent, name)}
+	parent.vpcs = append(parent.vpcs, &vpc)
+	return &vpc
+}
+func (tn *VpcTreeNode) children() ([]SquareTreeNodeInterface, []IconTreeNodeInterface, []LineTreeNodeInterface) {
+	return append(tn.zones, tn.sgs...), tn.elements, tn.connections
+}
+func (tn *VpcTreeNode) IsVPC() bool { return true }
+
+///////////////////////////////////////////////////////////////////////
+
+type ZoneTreeNode struct {
+	abstractSquareTreeNode
+	subnets []SquareTreeNodeInterface
+}
+
+func NewZoneTreeNode(parent *VpcTreeNode, name string) *ZoneTreeNode {
+	zone := ZoneTreeNode{abstractSquareTreeNode: newAbstractSquareTreeNode(parent, name)}
+	parent.zones = append(parent.zones, &zone)
+	return &zone
+}
+func (tn *ZoneTreeNode) children() ([]SquareTreeNodeInterface, []IconTreeNodeInterface, []LineTreeNodeInterface) {
+	return tn.subnets, tn.elements, tn.connections
+}
+func (tn *ZoneTreeNode) IsZone() bool { return true }
+
+///////////////////////////////////////////////////////////////////////
+
+type SGTreeNode struct {
```

Comment:
I made it simpler:
1. the parent of partialGS in the tree is SG
2. I added to partialGSTreeNode the method DrawioParentID() to return the zone ID
3. I added NotShownInDrawio() method to all treeNodes
4. added relevant comments

Reply:
I'm not sure I got it.
The parent of a security group should be the VPC as it can cross zones. The code seems right here.
The (drawio) parent of a PartialSecurityGroup should be a zone. But I don't understand how the code gives you the zone. `tn.Parent().Parent().ID()` should give you the VPC again.

Reply:
in the tree:
the parent of SG is VPC.
the parent of PartialSG is SG

In drawio:
that drawio parent  of PartialSG is VPC, (not a zone) (PartialSG can be on different zones)
the code follow these.

Reply:
ok, now I understand. your previous comment confused me.